src/web/aws-cloudwatch/reducers/StepsContext.jsx

```diff
@@ -0,0 +1,60 @@
+
+import React, { createContext, useReducer } from 'react';
+import PropTypes from 'prop-types';
+
+import {
+  ADD_ENABLED_STEP,
+  SET_CURRENT_STEP,
+  SET_AVAILABLE_STEPS,
+} from '../../common/constants';
+
+const StepsContext = createContext();
+
+const StepsProvider = ({ children }) => {
+  const initialState = {
+    availableSteps: [],
+    currentStep: 'authorize',
+    enabledSteps: ['authorize'],
+  };
+
+  const reducer = (state, action) => {
+    switch (action.type) {
+      case SET_CURRENT_STEP: {
+        return {
+          ...state,
+          currentStep: action.value,
+        };
+      }
+
+      case ADD_ENABLED_STEP: {
+        return {
+          ...state,
+          enabledSteps: [...state.enabledSteps, action.value],
+        };
+      }
+
+      case SET_AVAILABLE_STEPS: {
+        return {
+          ...state,
+          availableSteps: action.value,
+        };
+      }
+
+      default: return state;
+    }
+  };
+
+  const [state, dispatch] = useReducer(reducer, initialState);
```

Comment:
In this case we can also decouple those three, independent variables into separate states like this:

```
const [currentStep, setCurrentStep] = useState('authorize');
const [enabledSteps, enableStep] = useState(['authorize']);
const [availableSteps, makeStepsAvailable] = useState([]);

[...]
<StepsContext.Provider value={{ currentStep, setCurrentStep, enabledSteps, enableStep, availableSteps, makeStepsAvailable }}>
```

This way those variables are handled independently and we have already have those rich, abstracted (in contrast to `state`/`dispatch`) setters that are now created in `hooks/step.js`. We could even separate the `StepsContext` into three smaller contexts if we want to.

Even if we decide that those variables need to be coupled at some point, we can still do it by creating and exporting a different setter that cascades updates, or even switch back to a reducer again and create and export the same setters again.